frontend/src/TopBar.js

```diff
@@ -0,0 +1,130 @@
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import {
+  AppBar,
+  Toolbar,
+  makeStyles,
+  Typography,
+  Button,
+  IconButton,
+  MenuItem,
+  Drawer,
+  Icon,
+  Divider
+} from '@material-ui/core';
+import {AccountCircle, Menu} from '@material-ui/icons'
+import {Route,BrowserRouter as Router} from 'react-router-dom';
+import {ThemeProvider,createMuiTheme} from '@material-ui/core/styles'
+import {green} from '@material-ui/core/colors'
+import {colors } from '@material-ui/core';
+import {Link} from 'react-router-dom';
+
+
+const useStyles = makeStyles(({
+  root: {},
+  toolbar: {
+    height: 64
+  }
+}));
+
+const theme = createMuiTheme({
+  palette:{
+    primary:{
+      main: green[500]
+    },
+    secondary: {
+      main: colors.lightBlue[500]
+    },
+    text: {
+      primary: colors.blueGrey[900],
+      secondary: colors.blueGrey[600]
+    }
+  },
+
+  typography :{
+    h1:{
+      fontFamily: "sans-serif",
+      fontSize:"1rem"
+    },
+    h3:{
+    fontFamily: "sans-serif",
+    fontSize:"1.3rem"
+    }
+  }
+
+  }
+)
+
+
+const TopBar = () => {
+  const [open,setOpen] = useState(false)
+  const handleDrawer=()=>{
+    setOpen(true)
+  }
+  const classes = useStyles();
+  return (
+    <ThemeProvider theme = {theme}> 
+    <Router>
+    <div>
+      <AppBar position='static' color='primary'>  
+        <Toolbar>
+            
+            <IconButton onClick ={handleDrawer} color='inherit' edge='start' aria-label='Menu'>
+              <Menu>
+              </Menu>
+            </IconButton>
+            <Typography align='left' variant = 'h3' color='inherit' style={{flexGrow:1}}>
+              PLAYBUDDY
+            </Typography>
+            <Button component ={Link} to="/Login" color='inherit' align ='right'>
+              LOGIN
+            </Button>
+            <Button component = {Link} to="/Register" color='inherit' align = 'right'>
+              REGISTER
+            </Button>
+            <IconButton component = {Link} to="/UserActivity" color='inherit' align = 'right' >
+              <AccountCircle/> 
+              </IconButton>
+        </Toolbar>
+     </AppBar>    
+     <Drawer
+       anchor ='left'
+       open ={open}
+       onClose ={() => setOpen(false)}
+       >
+      <div style={{height : '100%', width : '250px'}}>
+        <MenuItem component ={Link} to="/" >
+         <IconButton size='small'>
+          Post
+         </IconButton>
+        </MenuItem>
+        <MenuItem component ={Link} to="/Activity" color='inherit' align ='right'>
```

Comment:
I mentioned this in another comment, but I'm not sure specifying the `component` property will work as expected in this case. It might be more straightforward to use the history hooks to change location.